Added documentation for nbs indexes. (#92)
NUbots/NUbook: src/book/02-system/01-foundations/04-nuclear.mdx

All values within this format are little endian.

> An `nbz` or `nbs.gz` file is a compressed `nbs` file using gzip.

Reviewer: ```suggestion
An `.nbz` or `.nbs.gz` file is a compressed `nbs` file using gzip.
```

Author: Do we want to put a dot here? Above, for normal `nbs` files, we don't have a dot.

> NBS files can have an acsossiated index file. This allows for quicker random access to the coresponding nbs file. An `nbs.idx` file has the folowwing frames repeated continuously, which are then compressed with gzip:

Reviewer: ```suggestion
NBS files can have an associated index file. This allows for quicker random access to the coresponding nbs file. An `.nbs.idx` file has the following frames repeated continuously, which are then compressed with gzip:
```

Also, is this saying that the .idx files are compressed already, or is compression something you have to do after you make the idx file?

| Name      | Type   | Description                                                                      |
| --------- | ------ | -------------------------------------------------------------------------------- |

> | hash      | uint64 | the 64bit hash for the payload type                                              |

Reviewer: ```suggestion
| hash      | uint64 | The 64-bit hash for the payload type                                              |
```

Reviewer: What type of hash is it? Md5 or something?

> | subtype   | uint32 | the id field of the message if it exists, otherwise 0                            |

Reviewer: ```suggestion
| subtype   | uint32 | The `id` of the message if it exists, otherwise 0                            |
```

> | timestamp | uint64 | timestamp field of the message if it exists or the emit timestamp in nanoseconds |

Reviewer: ```suggestion
| timestamp | uint64 | The `timestamp` of the message if it exists or the emit timestamp in nanoseconds |
```
What's the emit timestamp relative to? (Although I don't know if this is the right spot to talk about that?)

> | offset    | uint64 | offset to start of radiation symbol ☢                                            |

Reviewer: ```suggestion
| offset    | uint64 | Offset to start of radiation symbol ☢                                            |
```

> | size      | uint32 | Size of the whole packet from the radiation symbol                               |

Reviewer: Including the radiation symbol?

> This index file is also little enidan.

Reviewer: ```suggestion
This index file is little endian.
```

Reviewer: Also, wdym by "this" index file? are there others? This begs the question

> All values within the index file is little endian.

Reviewer: ```suggestion
All values within the index file are little endian.
```
